PayloadPkg/FirmwareUpdate/FirmwareUpdateHelper.c

```diff
@@ -584,6 +585,56 @@ IsUpdateComponentForContainer (
    return TRUE;
 }
 
+/**
+  Perform Config data svn checks
+
+  This function will perform svn checks for cfgdata in flash and
+  config data in capsule.
+
+  @param[in]  ImageHdr       Pointer to fw mgmt capsule Image header
+  @param[out] SvnStatus      Svn compare status
+
+  @retval  EFI_SUCCESS      SVN check successful.
+  @retval  other            error occurred during firmware update
+**/
+EFI_STATUS
+CheckSblConfigDataSvn (
+  IN   EFI_FW_MGMT_CAP_IMAGE_HEADER  *ImageHdr,
+  OUT  UINT8                         *SvnStatus
+  )
+{
+  CDATA_BLOB_ON_FLASH       *CfgBlobCapAddr;
+  CDATA_BLOB_ON_FLASH        CfgBlobFlashAddr;
+  UINT32                     CfgBlobSize;
+  EFI_STATUS                 Status;
+
+  ZeroMem (&CfgBlobFlashAddr, sizeof(CfgBlobFlashAddr));
+  Status = GetComponentInfo (FLASH_MAP_SIG_CFGDATA, (UINT32 *)&CfgBlobFlashAddr, &CfgBlobSize);
+  if (EFI_ERROR (Status)) {

```

Comment:
SBL CFG data has two copies in BP0 and BP1. GetComponentInfo () will read current CFG data copy. 
I think here it should read the other CFG data copy since the firmware update is updating the CFG data of the other copy.

Reply:
Are these copies of CFGDATA in Redundant A and B are same or different?  I expect it to be same.
The flow looks as, first its checks SVN for CFGDATA in Redundant A and it updates Redundant B CFGDATA.
Next time its check CFGDATA in  Redundant B. As this block is already updated, SVN check passes and updated Redundant A.

Reply:
When updating CFG data B, the SVN from current boot CFG data A is correct since both A and B has same value (e.g. 2), after CFG B SVN is updated to new value (e.g. 3), and boot with CFG data B, the SVN from current boot CFG data would be new SVN (3.)in our logic. 
Ideally, regardless booting from A or B, both should get same SVN number (old one) and has exact same behavior.